scripts/chroot-xfce.sh

```diff
@@ -0,0 +1,291 @@
+        if [[ $i -eq $selected ]]; then
+            echo -ne " ${highlight}> ${options[$i]} ${normal}\n"
+        else
+            echo -ne "   ${options[$i]}\n"
+        fi
+    done
+
+    echo -e "\n${normal}(Use Up/Down arrows, Enter, Escape, q, Home, or End)"
+}
+
+# Function to draw the distro selection menu
+draw_distro_menu() {
+    clear
+    echo -e "\n${logolight}${logo}"
+    echo -e "\n${normal}Login to Distro"
+    echo -e "${error}${formatted_distro}\n"
+    echo -e "${normal}${infoa[RANDOM % ${#infoa[@]}]}"
+    echo -e "${normal}${infob[RANDOM % ${#infob[@]}]}\n"
+
+    echo -e "${title}-- Select Distro --${normal}"
+    for i in "${!installed_distros[@]}"; do
+        if [[ $i -eq $selected_distro_index ]]; then
+            echo -ne " ${highlight}> ${installed_distros[$i]} ${normal}\n"
+        else
+            echo -ne "   ${installed_distros[$i]}\n"
+        fi
+    done
+
+    echo -e "\n${normal}(Use Up/Down arrows, Enter to select, Escape or q to exit)"
+}
+
+# Function to select distro
+select_distro() {
+    selected_distro_index=0
+    distro_selected=false
+
+    draw_distro_menu
+
+    while true; do
+        stty -echo
+        read -r -sN1 char
+        case "$char" in
+            $'\e')
+                read -r -sN2 -t 0.1 char2
+                case "$char2" in
+                    '[A') # Up
+                        selected_distro_index=$(( (selected_distro_index - 1 + ${#installed_distros[@]}) % ${#installed_distros[@]} ))
+                        ;;
+                    '[B') # Down
+                        selected_distro_index=$(( (selected_distro_index + 1) % ${#installed_distros[@]} ))
+                        ;;
+                    '[H') # Home
+                        selected_distro_index=0
+                        ;;
+                    '[F') # End
+                        selected_distro_index=$((${#installed_distros[@]} - 1))
+                        ;;
+                    *)
+                        stty echo
+                        echo -e "$show_cursor"
+                        return 1
+                        ;;
+                esac
+                ;;
+            $'\n'|$'\r') # Enter
+                selected_distro="${installed_distros[$selected_distro_index]}"
+                export distro_selected=true
+                stty echo
+                return 0
+                ;;
+            $'\x03'|'q') # Ctrl+C or q
+                stty echo
+                echo -e "$show_cursor"
+                exit 0
+                ;;
+        esac
+        draw_distro_menu
+    done
+}
+
+# Function to unmount remaining mounted points
+unmount_chroot() {
+  local chmount="$chrootdistro_dir/$selected_distro"
+
+  sudo chroot-distro unmount "$selected_distro" &> /dev/null
+
+  local leftover=(
+    "$chmount/data"
+    "$chmount/tmp"
+    "$chmount/dev"
+  )
+
+  for mount_point in "${leftover[@]}"; do
+    if [[ -d "$mount_point" ]]; then
+      su -c "umount -l '$mount_point'" 2>/dev/null
```

Comment:
You shouldn't use `-l` as it may cause resource leak. It should be used only as a last resort when nothing else works.